### PR TITLE
[CLI-435] fix loading config yaml files

### PR DESCRIPTION
**Pull request by harukatab**
- parse and load multiple config files that are set through env variable (e.g. `export WANDB_CONFIG_PATHS=this-config.yaml,that-config.yaml`)
- automatically load "config-defaults.yaml"
- take a string file path to the config file as config argument in init function (e.g. `wandb.init(config=filename)`)

### CODE REVIEW

File: wandb/sdk/wandb_setup.py [hunk 1]
```diff
@@ -202,9 +202,16 @@ def _setup(self):
         # if config_paths was set, read in config dict
         if self._settings.config_paths:
             # TODO(jhr): handle load errors, handle list of files
-            self._config = config_util.dict_from_config_file(
-                self._settings.config_paths
-            )
+            config_paths = self._settings.config_paths.split(",")
+            for config_path in config_paths:
+                print(config_path)
```
Comment (raubitsj) [inline wandb/sdk/wandb_setup.py:207]: debugging left in code

File: wandb/sdk/wandb_helper.py [hunk 2]
```diff
@@ -4,20 +4,21 @@
...
 from wandb.errors.error import UsageError
+from wandb.lib import config_util
 
 
 def parse_config(params, exclude=None, include=None):
     if exclude and include:
         raise UsageError("Expected at most only one of exclude or include")
-
+    if isinstance(params, str):
+        params = config_util.dict_from_config_file(params)
```
Comment (raubitsj) [inline wandb/sdk/wandb_helper.py:14]: isnt params going to get blown away in the next line

File: wandb/sdk/wandb_setup.py (hunk 1 above)
Comment (raubitsj) [inline wandb/sdk/wandb_setup.py:207]: left in debugging line

File: wandb/sdk/wandb_helper.py (hunk 2 above)
Comment (raubitsj) [reply wandb/sdk/wandb_helper.py:14]: and looks like we just get a log (debug level ) that the file didnt exist, but i htink the wandb.init() call should fail

**Review by raubitsj — CHANGES_REQUESTED** [review]
Looks pretty good, some minor issues in previous comments:

File: tests/wandb_config_test.py [hunk 3]
```diff
@@ -28,3 +30,14 @@ def test_locked_set():
     s.update_locked(dict(this=2, that=4), "sweep")
     s.this = 8
     assert s.this == 2
+
+
+def test_load_config_default():
+    test_path = os.path.join("config-defaults.yaml")
```
Comment (farizrahman4u) [inline tests/wandb_config_test.py:36]: Nit pick: Here `os.path.join()` is a No-op. `test_path = "config-defaults.yaml"` would do.

File: wandb/lib/config_util.py [hunk 4]
```diff
@@ -44,16 +48,20 @@ def save_config_file_from_dict(config_filename, config_dict):
 
 
 def dict_from_config_file(config_filename):
+    file_path = os.path.join(config_filename)
```
Comment (farizrahman4u) [inline wandb/lib/config_util.py:51]: Same here.

**Review by farizrahman4u — CHANGES_REQUESTED** [review]

File: wandb/sdk/wandb_helper.py (hunk 2 above)
Comment (raubitsj) [reply wandb/sdk/wandb_helper.py:14]: ignore the "params blown away" comment, file not found needs still needs to be addressed.

File: wandb/lib/config_util.py [hunk 5]
```diff
@@ -43,17 +47,22 @@ def save_config_file_from_dict(config_filename, config_dict):
...
     except OSError:
-        raise ConfigError("Couldn't read config file: %s" % config_filename)
+        raise ConfigError("Couldn't read config file: %s" % filename)
     try:
         loaded = load_yaml(conf_file)
     except yaml.parser.ParserError:
         raise ConfigError("Invalid YAML in config yaml")
     config_version = loaded.pop("wandb_version", None)
-    if config_version != 1:
+    if config_version and config_version != 1:
```
Comment (farizrahman4u) [inline wandb/lib/config_util.py:65]: `if config_version is not None and config_version != 1` (so that we raise if `config_version` is 0 as well)

Comment (harukatab) [reply wandb/lib/config_util.py]: Thank you for all the feedback!!!

**Review by adrnswanberg — APPROVED** [review]

File: wandb/sdk/wandb_helper.py [hunk 6]
```diff
@@ -4,20 +4,21 @@
...
 import six
 from wandb.errors.error import UsageError
+from wandb.lib import config_util
 
 
 def parse_config(params, exclude=None, include=None):
     if exclude and include:
         raise UsageError("Expected at most only one of exclude or include")
-
+    if isinstance(params, str):
```
Comment (raubitsj) [inline wandb/sdk/wandb_helper.py:13]: You need to use six here.
```
import six
isinstance(s, six.string_types)
```

**Review by raubitsj — CHANGES_REQUESTED** [review]

**Pull request merged**